vaccine_feed_ingest/runners/wi/arcgis_map/normalize.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+
+import datetime
+import json
+import os
+import pathlib
+import re
+import sys
+from typing import List, Optional
+
+from arcgis.geometry import project
+from arcgis.gis import GIS
+from vaccine_feed_ingest_schema import location as schema
+
+from vaccine_feed_ingest.utils.log import getLogger
+
+logger = getLogger(__file__)
+
+GIS()  # initialize GIS, for _get_locations_batch
+
+
+def _get_id(site: dict) -> str:
+    data_id = site["attributes"]["OBJECTID"]
+
+    # Could parse these from directory traversal, but do not for now to avoid
+    # accidental mutation.
+    site_name = "arcgis_map"
+    runner = "wi"
+
+    # Could parse these from the input file name, but do not for now to avoid
+    # accidental mutation.
+    arcgis = "wi_arcgis_map"
+    layer = 0
+
+    return f"{runner}_{site_name}:{arcgis}_{layer}_{data_id}"
+
+
+def _get_contacts(site: dict) -> Optional[List[schema.Contact]]:
+    if website := site["attributes"]["WEBSITE_URL"]:
+        if re.match(r"^https://(www\.)?google\.com/search\?q=", website):
+            website = website.replace(" ", "+")
+
+        if website == "https://wildwoodfamilyclinic/":
+            website = "https://wp.wildwoodclinic.com/"
+
+        # workaround until samuelcolvin/pydantic#2778 is merged
+        website = website.rstrip("#")
+
+        return [schema.Contact(website=website)]
+
+    return None
+
+
+# The input geometry is in a different projection from our normalized geometry,
+# so we need to project it. The projection api from arcgis has a noticeable
+# per-call overhead, so we batch the projections.
+def _get_locations_batch(sites: List[dict]) -> List[Optional[schema.LatLng]]:
+    result = []
+    for x in project([s["geometry"] for s in sites], in_sr=3857, out_sr=4326):
```

Comment:
Can you use `pyproj` for re-projecting geometries instead of Arcgis: https://pypi.org/project/pyproj/

I think it will be much faster, and isn't as heavyweight as arcgis

Reply:
Much better. Thanks for the suggestion!